crowbar_framework/app/models/api/node.rb

```diff
@@ -72,7 +69,26 @@ def os_upgrade
         Rails.logger.error("Executing of os upgrade script has failed on node #{@node.name}.")
         return false
       end
-      true
+
+      begin
```

Comment:
There's a lot of duplicated code here.  Why not have a helper which calls `@node.ssh_cmd` and then `@node.wait_for_script_to_finish`?  You would just need to pass parameters for the script name, description, and failure / success messages.